src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -1011,10 +1011,11 @@ private MemberSemanticModel CreateMemberModel(CSharpSyntaxNode node)
 
                 case SyntaxKind.GlobalStatement:
                     {
-                        Debug.Assert(!this.IsRegularCSharp);
                         var parent = node.Parent;
                         // TODO (tomat): handle misplaced global statements
```

Comment:
I think it is still relevant as we don't support SemanticModel for global code if it is not part of a Script. Nothing changed in that regard.